From 42a5ccf0f6e42c1487fef1d5422939bc6836044d Mon Sep 17 00:00:00 2001
From: Andy Nichols <andy.nichols@nokia.com>
Date: Thu, 24 Nov 2011 10:19:18 +0100
Subject: [PATCH 1/2] Removed superfluous call to wl_copy_connection

The only purpose those code seems to serve is to introduce a buffer
overflow when events contain more than 128 bytes of data.
---
 src/wayland-client.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/src/wayland-client.c b/src/wayland-client.c
index 939c17d..fd5a8fa 100644
--- a/src/wayland-client.c
+++ b/src/wayland-client.c
@@ -445,12 +445,10 @@ static void
 handle_event(struct wl_display *display,
 	     uint32_t id, uint32_t opcode, uint32_t size)
 {
-	uint32_t p[32];
 	struct wl_proxy *proxy;
 	struct wl_closure *closure;
 	const struct wl_message *message;
 
-	wl_connection_copy(display->connection, p, size);
 	proxy = wl_map_lookup(&display->objects, id);
 
 	if (proxy == WL_ZOMBIE_OBJECT) {
-- 
1.7.8.rc2

